src/state/controls/Button.ts

```diff
@@ -71,4 +71,11 @@ export class Button extends Control<IButtonData> implements IButton {
     public giveInput(input: IButtonInput): Promise<void> {
         return this.sendInput(input);
     }
+
+    public update(changedData: Partial<IButtonData>): Promise<void> {
+        if (changedData.cooldown) {
```

Comment:
Ok, so:

before: "I want to set the button's text to "connor" and the cooldown to 5 seconds"
```
control.setText('connor').then(() => control.setCooldown(5))
```

After:
```
control.update({
 text:'connor',
 cooldown:<A utc what?>
});
```